oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtIssuerAuthenticationManagerResolver.java

```diff
@@ -85,6 +85,27 @@ public JwtIssuerAuthenticationManagerResolver(Collection<String> trustedIssuers)
 		Assert.notEmpty(trustedIssuers, "trustedIssuers cannot be empty");
 		this.issuerAuthenticationManagerResolver = new TrustedIssuerJwtAuthenticationManagerResolver(
 				Collections.unmodifiableCollection(trustedIssuers)::contains);
+		this.issuerConverter = new JwtClaimIssuerConverter();
+	}
+
+	/**
+	 * Construct a {@link JwtIssuerAuthenticationManagerResolver} with a custom
+	 * {@link JwtAuthenticationConverter} using the provided parameters
+	 *
+	 * A custom {@link JwtAuthenticationConverter} allows to use a custom
+	 * {@link Converter} (much like {@link JwtGrantedAuthoritiesConverter}) to handle an
+	 * untypical JWT token
+	 * @param trustedIssuers a list of trusted issuers
+	 * @param jwtAuthenticationConverter a custom {@link JwtAuthenticationConverter}
+	 * @since 5.4
+	 */
+	public JwtIssuerAuthenticationManagerResolver(Collection<String> trustedIssuers,
+			JwtAuthenticationConverter jwtAuthenticationConverter) {
```

Comment:
While this change could be valuable, let's please keep this PR focused on #8535, which is about customizing the `BearerTokenResolver`. Would you please remove these changes and create a ticket for this separate feature?